src/KeyVault/KeyVault/Models/PSKeyVaultCertificatePolicy.cs

```diff
@@ -304,6 +452,261 @@ internal static PSKeyVaultCertificatePolicy FromCertificatePolicy(CertificatePol
+                Exportable = certificatePolicy.Exportable == null ? null : certificatePolicy.Exportable,
+                ReuseKeyOnRenewal = certificatePolicy.ReuseKey == null ? null : certificatePolicy.ReuseKey,
+                SubjectName = certificatePolicy.Subject == null ? null : certificatePolicy.Subject,
+                DnsNames = certificatePolicy.SubjectAlternativeNames == null || certificatePolicy.SubjectAlternativeNames.DnsNames == null ?
+                    null : new List<string>(certificatePolicy.SubjectAlternativeNames.DnsNames),
+                Emails = certificatePolicy.SubjectAlternativeNames == null || certificatePolicy.SubjectAlternativeNames.Emails == null ?
+                    null : new List<string>(certificatePolicy.SubjectAlternativeNames.Emails),
+                UserPrincipalNames = certificatePolicy.SubjectAlternativeNames == null || certificatePolicy.SubjectAlternativeNames.UserPrincipalNames == null ?
+                    null : new List<string>(certificatePolicy.SubjectAlternativeNames.UserPrincipalNames),
+                KeyUsage = certificatePolicy.KeyUsage == null ? null : certificatePolicy.KeyUsage == null ? null : certificatePolicy.KeyUsage.Select(keyUsage => keyUsage.ToString()).ToList(),
+                Ekus = certificatePolicy.EnhancedKeyUsage == null ? null : new List<string>(certificatePolicy.EnhancedKeyUsage),
+                ValidityInMonths = certificatePolicy.ValidityInMonths == null ? null : certificatePolicy.ValidityInMonths,
+                CertificateTransparency = certificatePolicy.CertificateTransparency == null ? null : certificatePolicy.CertificateTransparency,
+                IssuerName = certificatePolicy.IssuerName == null ? null : certificatePolicy.IssuerName,
+                CertificateType = certificatePolicy.CertificateType == null ? null : certificatePolicy.CertificateType,
+                RenewAtNumberOfDaysBeforeExpiry = certificatePolicy.LifetimeActions == null ? null : FindIntValueForAutoRenewAction(certificatePolicy.LifetimeActions),
+                RenewAtPercentageLifetime = certificatePolicy.LifetimeActions == null ? null : FindIntValueForAutoRenewAction(certificatePolicy.LifetimeActions),
+                EmailAtNumberOfDaysBeforeExpiry = certificatePolicy.LifetimeActions == null ? null : FindIntValueForEmailAction(certificatePolicy.LifetimeActions),
+                EmailAtPercentageLifetime = certificatePolicy.LifetimeActions == null ? null : FindIntValueForEmailAction(certificatePolicy.LifetimeActions),
+                Enabled = certificatePolicy.Enabled == null ? null : certificatePolicy.Enabled,
+                Created = certificatePolicy.CreatedOn.HasValue ? certificatePolicy.CreatedOn.Value.DateTime : (DateTime?)null,
+                Updated = certificatePolicy.UpdatedOn.HasValue ? certificatePolicy.UpdatedOn.Value.DateTime : (DateTime?)null,
+            };
+        }
+        private void ReadKeyProperties(JsonElement json)
```

Comment:
For your reference, https://github.com/Azure/azure-powershell/blob/58b5a0bdbb7c84eb30b651983affb3652c3384d3/src/KeyVault/KeyVault/Commands/Key/KeyRotation/SetAzKeyVaultKeyRotationPolicy.cs#L150. please compare two ways from json file to object and pick a simper way.

Reply:
The original way help avoiding check if unnecessary properties nullable. And wrapped the properties by blocks. 
The referred code accepts mutiple actions added. But for `PSKeyVaultCertificatePolicy`, only one of the values for RenewAtNumberOfDaysBeforeExpiry, RenewAtPercentageLifetime, EmailAtNumberOfDaysBeforeExpiry, EmailAtPercentageLifetime  can be set. Hence the complexity is not the same.